src/backend/drm/atomic/surface.rs

```diff
@@ -44,7 +44,7 @@ pub(super) struct AtomicDrmSurfaceInternal<A: AsRawFd + 'static> {
     pub(super) state: RwLock<State>,
     pub(super) pending: RwLock<State>,
     pub(super) logger: ::slog::Logger,
-    pub(super) init_buffer: Cell<Option<(DumbBuffer, framebuffer::Handle)>>,
+    pub(super) test_buffer: Cell<Option<(DumbBuffer, framebuffer::Handle)>>,
```

Comment:
What is the role of this `test_buffer` ?

Reply:
Atomic commits without a framebuffer fail for an active crtc, even if it is just a test-commit. (Because test-commits just simulate a real-commit.) Thus we need a framebuffer, which we can use in the various test-commits, that we do to check each state-change beforehand. (Like adding or removing a connector.)